src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -393,6 +393,7 @@ boolean populateDefaults() throws IOException, GeneralSecurityException {
                 // the hostname or ip/port that is only available within the
                 // cluster
                 this.defaultedRedirectURL = provider.issuer;
+                this.testProviderEndpoints(credential, transport);
```

Comment:
this doesn't check the return code so what's it doing?

Reply:
it is a diagnostic,  given what @enj mentioned about the oauth stuff moving around, coming up and down ... I can add a comment